### PR TITLE
Require authentication of returning member before checkout

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Require login before checkout

This is the first step with very quick and dirty implementation to unblock signing

## Justify why they are needed

Handle returning members

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1443** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1443" target="_blank">Graphite</a> 
    * **PR #1446** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1446" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1446?utm_source=stack-comment).

File: apps/store/src/components/CartPage/CartPage.tsx [hunk 1]
```diff
@@ -95,6 +104,79 @@ const EmptyState = ({ children, prevURL }: EmptyStateProps) => {
...
+    return (
+      <BankIdLogin
+        shopSessionId={shopSession.id}
+        ssn={shopSession.customer.ssn}
+        onComplete={handleAuthSuccess}
+      />
+    )
+  } else {
+    return (
+      <ButtonNextLink href={PageLink.checkout()} onClick={reportCheckout}>
```
Comment (robinandeer) [inline apps/store/src/components/CartPage/CartPage.tsx:137]: nice to handle reporting here 😃

File: apps/store/src/components/CartPage/CartPage.tsx [hunk 2]
```diff
@@ -95,6 +104,79 @@ const EmptyState = ({ children, prevURL }: EmptyStateProps) => {
...
+// TODO: Make it reusable between cart and price calculator prompt
+// TODO: Localize texts
+const BankIdLogin = (props: { shopSessionId: string; ssn: string; onComplete: () => void }) => {
+  const [state, setState] = useState<'IDLE' | 'PROGRESS' | 'ERROR'>('IDLE')
+  const { shopSessionId } = props
+  const [authenticateShopSession] = useShopSessionAuthenticateMutation({
+    variables: { shopSessionId },
+  })
+  const onLogin: MouseEventHandler = async (event) => {
+    event.preventDefault()
```
Comment (robinandeer) [inline apps/store/src/components/CartPage/CartPage.tsx:153]: what are we preventing here?

File: apps/store/src/components/CartPage/CartPage.tsx [hunk 3]
```diff
@@ -95,6 +104,79 @@ const EmptyState = ({ children, prevURL }: EmptyStateProps) => {
...
+    }
+  }
+  return (
+    <>
+      <Text>
+        Looks like you&quot;re returning member. You need to login before proceeding to checkout
+      </Text>
+      <Button onClick={onLogin} loading={state !== 'IDLE'}>
+        BankID Login
+      </Button>
```
Comment (robinandeer) [inline apps/store/src/components/CartPage/CartPage.tsx:173]: nit: This should be a form to allow screen readers to see that there are actions on the page

**Review by robinandeer — APPROVED** [review]
Awesome! Excited to try it out

**Pull request merged**